From bfdcfd3b8c8207f642ad68a6f9bdf8e2e9997353 Mon Sep 17 00:00:00 2001
From: Makoto Onuki <omakoto@google.com>
Date: Fri, 7 Sep 2018 11:32:47 -0700
Subject: [PATCH 1/3] Don't use WAL in TelephonyProvider to work around
 b/113352727

This is a partial cherry-pick of the change Idd508dfb6229 in master

============================
The regular journal mode (non-wal) is less performant on ext4
(i.e. pixel 1 / 2) but actually more performant on F2FS
(i.e. pixel 3)

Bug: 114241235
Bug: 115990430
Test: Manual test:
- Receive a call
- Open Messenger app
- Run "dumpsys dbinfo com.android.phone"
- Make sure useCompatibilityWal is all "false"

Test: pi-dev cherry-pick: Manual test
- Send a text, receive a text

Test: am instrument -w com.android.providers.telephony.tests

Change-Id: Icdf16da37f272703cb782643cb7d510924d58a89
Merged-in: Idd508dfb6229a17e364d508159f1ea94a9804634
(cherry picked from commit 5a29f7bfdc018f4645a64492784efb85d3075bbe)
(cherry picked from commit e2d75c352911569b7de110e7f378a5ef24d95738)
---
 src/com/android/providers/telephony/MmsSmsDatabaseHelper.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/providers/telephony/MmsSmsDatabaseHelper.java b/src/com/android/providers/telephony/MmsSmsDatabaseHelper.java
index 04ffa20..a7f10f6 100644
--- a/src/com/android/providers/telephony/MmsSmsDatabaseHelper.java
+++ b/src/com/android/providers/telephony/MmsSmsDatabaseHelper.java
@@ -252,6 +252,7 @@ public class MmsSmsDatabaseHelper extends SQLiteOpenHelper {
         mContext = context;
         // Memory optimization - close idle connections after 30s of inactivity
         setIdleConnectionTimeout(IDLE_CONNECTION_TIMEOUT_MS);
+        setWriteAheadLoggingEnabled(false);
     }
 
     /**
-- 
2.17.1

